### ISSUE TITLE
License, CLA

### ISSUE THREAD

**Issue by davidak** [domenkozar/nix.dev#46]
The used license is more restrictive than the GPL and not even considered open source. It don't fit the ecosystem and is not compatible with the license of the nixos website.

I suggest to use the same license as the website (CC BY-SA) since the plan is to integrate the content there.

And what is the plan with the CLA? Do you want to get rich by selling the tutorials as book or add a paywall later?

I don't want to give you the copyright but share it freely with the community, so anyone could sell locally printed books if there is demand for that. That's better for the environment and more fair.

**Comment by domenkozar** [comment]
> The used license is more restrictive than the GPL and not even considered open source. 

GPL is a license or code, CC is for content. I can see you added CC yourself to nixos homepage.

> I suggest to use the same license as the website (CC BY-SA) since the plan is to integrate the content there.

It is the same license with an additional restriction of not being able to make money from this content.

> And what is the plan with the CLA? Do you want to get rich by selling the tutorials as book or add a paywall later?

Even if I would want add a paywall, the license allows anyone to host this documentation anywhere they like, as long as they don't charge for it. 

If it's freely available, why would I paywall it?

CLA does allow me to be able to license it under different terms, if I decide to make it a book.

> I don't want to give you the copyright but share it freely with the community, 

 Note that the license allows you to share it freely.

> so anyone could sell locally printed books if there is demand for that. That's better for the environment and more fair.

Quite on the contrary, it's close to impossible (as far as I know) to find a publisher if you don't own the copyright, so noone would be able to sell the books.

If you don't wish to contribute the content under these terms, that's understandable and no harm feelings. I don't expect anyone to get rich with these either way.

**Comment by davidak** [comment]
>GPL is a license for source code, CC is for content.

sure, but you can compare the conditions

GPL has similar conditions like CC BY-SA while
MIT is similar to CC BY

when you add an additional restriction of not being able to make money from it, it would be considered proprietary since it violates requirements for free software resp. open source

https://www.gnu.org/licenses/license-list.html#NonFreeDocumentationLicenses
https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses

>I can see you added CC yourself to nixos homepage.

yes, and it was a huge effort that took more than 6 months! https://github.com/NixOS/nixos-homepage/pull/260 i want to prevent such license issues

>CLA does allow me to be able to license it under different terms, if I decide to make it a book.

yes, but why should you be the only person? isn't this a community project? without that restriction, you could still make a book and probably no one else would do it. if that is your concern

>Quite on the contrary, it's close to impossible (as far as I know) to find a publisher if you don't own the copyright, so noone would be able to sell the books.

i already bought books with CC license.

some examples i quickly found:

- https://en.wikipedia.org/wiki/List_of_major_Creative_Commons_licensed_works#Books
- https://www.transcript-verlag.de/978-3-8376-4732-7/raum-und-wuerde/ (CC BY)
- https://www.transcript-verlag.de/978-3-8376-5247-5/glueck-und-nachhaltigkeit/?c=313000000 (CC BY)

most use CC BY-NC-ND, but a more permissive license **is** possible

you can still introduce yourself as the author of the book to tho publisher, put "Domen Kožar and contributors" on the cover and name all contributors somewhere in the book. this way the publisher should be happy, amazon is happy that they only have to list one name instead of 50 in the book listing and the CC license is fulfilled (attribution)

**Comment by domenkozar** [comment]
> when you add an additional restriction of not being able to make money from it, it would be considered proprietary since it violates requirements for free software resp. open source

That argument is valid for code, but not for written text. I do not want to go into this debate, but let's say it's highly philosophical and I want to focus my energy on fixing Nix documentation, not defending different shades of freedom.

Please come up with concrete examples how it limits your freedom. If you have a plan to publish this as a book, we can talk about it why and how.

> yes, and it was a huge effort that took more than 6 months! NixOS/nixos-homepage#260 i want to prevent such license issues

And that's what I want to avoid with the CLA. 

> yes, but why should you be the only person? isn't this a community project? without that restriction, you could still make a book and probably no one else would do it. if that is your concern

No, because you need to publish under CC license AND hold copyright. The books you listed exist because those people hold copyright the same way as I will. Did you find a published book that copyright is held by 100 contributors?

It even says so in the GPL page you linked:

> In addition, it has a drawback for any sort of work: when a modified version has many authors, in practice getting permission for commercial use from all of them would become infeasible.

In pratice noncommercial restriction really doesn't matter, so I'm fine removing it.

CLA is however required if anybody ever wants to publish a book.

**Comment by davidak** [comment]
>That argument is valid for code, but not for written text. I do not want to go into this debate

that's fine, i respect that. i just wanted to point out my concern

>Please come up with concrete example

my main concern is that we would have to point out on nixos.org, that these tutorials have a different license than the rest of the site. when we use the same license, the statement from the website readme is enough an we can focus on fixing Nix documentation instead of licensing trouble

>In practice noncommercial restriction really doesn't matter, so I'm fine removing it.

that would be great and solves my main concern

>No, because you need to publish under CC license AND hold copyright.

ok, i don't understand why, but it doesn't matter

>CLA is however required if anybody ever wants to publish a book.

ok

**Comment by edibopp** [comment]
Thanks @davidak for pointing that out. I missed that when integrating the content, my bad.

I agree the CLA is valuable to stay in control over the content. If we want to turn this into official documentation though, would it not be better to use the NixOS foundation as copyright holder? I can imagine you (@domenkozar) may object in order to keep nix.dev autonomous. In that case we have to discuss whether it is a good idea to have this legal dependency on an individual.

**Comment by domenkozar** [comment]
> If we want to turn this into official documentation though, would it not be better to use the NixOS foundation as copyright holder? I can imagine you (@domenkozar) may object in order to keep nix.dev autonomous.

I am quite against that, since I'm quite tired of politics around ideologies. That is also why I started nix.dev as a seprate project in the first place, after weeks of arguing about documentation format.

>  In that case we have to discuss whether it is a good idea to have this legal dependency on an individual.

The lack of trust in NixOS community lately is quite saddening, but I'm not going to pick this fight but rather create content and let everyone decide for themselves if something worked for hundreds of hours and offered free is "good enough".

**Comment by domenkozar** [comment]
I'm going to change the license to CC BY-SA, since CC compatibility chart does say that NC is not complient with SA. That should make it trivial to integrate into nixos.org (I still think it's a waste of time that we could spend on tutorials, rathen than styling - which was motived that we just need to get things done).

About politics, feel free to bring CLA up on next marketing meeting and act accordingly.

**Comment by domenkozar** [comment]
a67a7485f77ab03dac852898ea2589b189c176c6

**Comment by edibopp** [comment]
When I expressed my concern about this legal issue, I wasn't thinking about trust – I probably should have. I am sorry for ignoring that implication.

This issue was brought up during the marketing meeting. We concluded that the status quo is fine. Regarding whether this integration is a waste of time or not… I believe we can disagree on this point without interfering with each other, which is what I intend to do from here on.
